src/main/java/com/musala/sdcs/devices/base/Device.java

```diff
@@ -0,0 +1,72 @@
+package com.musala.sdcs.devices.base;
+
+public abstract class Device implements IDevice {
+    private String manufacturer;
+    private String modelId;
+    private String serialNumber;
+    private String firmwareVersion;
+    private int hardwareVersion;
+    private String label;
+
+    public Device() {
+        
+    }
+    
+    public Device(String manufacturer, String modelId, String serialNumber, String firmwareVersion, int hardwareVersion,
+            String label) {
+        this.setManufacturer(manufacturer);
+        this.setModelId(modelId);
+        this.setSerialNumber(serialNumber);
+        this.setFirmwareVersion(firmwareVersion);
+        this.setHardwareVersion(hardwareVersion);
+        this.setLabel(label);
+    }
+
+    public String getManufacturer() {
```

Comment:
Add an `@Override` notation to all methods implemented from `IDevice` interface. Add JavaDoc to the class itself.